gptme/tools/shell.py

```diff
@@ -180,23 +180,30 @@ def _run(self, command: str, output=True, tries=0) -> tuple[int | None, str, str
         stdout = []
         stderr = []
         return_code = None
-        read_delimiter = False
 
         while True:
             rlist, _, _ = select.select([self.stdout_fd, self.stderr_fd], [], [])
+            if not rlist: continue
             for fd in rlist:
                 assert fd in [self.stdout_fd, self.stderr_fd]
                 # We use a higher value, because there is a bug which leads to spaces at the boundary
                 # 2**12 = 4096
                 # 2**16 = 65536
                 data = os.read(fd, 2**16).decode("utf-8")
-                re_returncode = re.compile(r"ReturnCode:(\d+)")
```

Comment:
Why no longer use this regex? I recently added it in https://github.com/ErikBjare/gptme/commit/df5c9d6eca2d3d084a1060f97290ad3f629930c0 and I think it fixed a bug. Not sure if it currently supports negative returncodes though.

Reply:
ahh I see, best to just leave it as is then. reverted